### PR TITLE
Feature/french translations

### LINKED ISSUE #13: [Localization] French/Français
Hello,
I'm a French player of Genshin impact and your app is a wonder.
I would like to know if any French localization would be possible. I can do most of the translation/check necessary if I have the localization files.
Thank you for your work and time.
Best Regards.

### PR DESCRIPTION
**Pull request by therealcorwin**
Hi :)

Sorry for the delay. I very busy. 

Complete rework for French Translation due to a vscode addon bug. 
==> Characters are OK. 
==> Need a second pass to the rest

Hope this time, the file format is correct ;)

Fixes #13

### CODE REVIEW

**Comment by Wolfteam** [comment]
Okay, will release a beta version with the french translations applied so you can test it.
Should be up in 2hrs ~

**Comment by therealcorwin** [comment]
Fix done :)

**Review by Wolfteam — COMMENTED** [review]
> Fix done :)

emm nothing has been fixed

**Comment by therealcorwin** [comment]
Oh OK. A little miss understanding. We are not talking about same things. 

I check this :)

**Comment by therealcorwin** [comment]
Hi @Wolfteam,

Do you need more information before merge the update file. Sorry, haven't i no time yet to make a second pass. I'm more available next week to translate 2.0 news and more

**Review by Wolfteam — COMMENTED** [review]
> Hi @Wolfteam,
> 
> Do you need more information before merge the update file ? Sorry, haven't i no time yet to make a second pass. I'm more available next week to translate 2.0 news and more

Emm no, just fix the comments that I left

File: assets/i18n/fr.json [hunk 1]
```diff
@@ -1681,2172 +1669,2311 @@
...
+          "description": "Augmentez le niveau de Fléau du mal de 3. Le niveau maximum est de 15.",
           "secondDescription": null,
           "descriptions": []
         },
-        {
+         {
           "key": "c6",
-          "title": "Conqueror of Evil: Guardian Yaksha",
-          "description": "Bane of All Evil, hitting at least 2 opponents with Xiao's Plunging Attack will immediately grant him 1 charge of Lemniscatic Wind Cycling, and for the next 1s, he may use Lemniscatic Wind Cycling while ignoring its CD.",
+          "title": "Dompteur de démons : Gardien YakshaLorsque Fléau du mal est actif, les attaques descendantes touchant au moins 2 ennemis confèrent immédiatement une charge de Cycle du vent lemniscatique supplémentaires, qui peut être activée dans la seconde qui suit, même s'il n'est pas entièrement rechargé.",
```
[the remark this answers is not in the document]
Comment (therealcorwin) [reply assets/i18n/fr.json:3724]: Done

[the remark this answers is not in the document]
Comment (Wolfteam) [reply assets/i18n/fr.json:3724]: nothing has been fixed...

[the remark this answers is not in the document]
Comment (therealcorwin) [reply assets/i18n/fr.json:3724]: On my local repo yes ;). I push all fix when it's done.

**Review by therealcorwin — COMMENTED** [review]
Change done.

**Comment by Wolfteam** [comment]
Left a few comments @therealcorwin

**Comment by therealcorwin** [comment]
Hi @Wolfteam,

I just check the file, and i don't see what i forgot :/

**Review by Wolfteam — COMMENTED** [review]
> Hi @Wolfteam,
> 
> I just check the file, and i don't see what i forgot :/

Check the review review comments
review

**Comment by therealcorwin** [comment]
Strange I don't see your review comment :/ I fix this in few minutes

**Comment by therealcorwin** [comment]
I just push some modifications

**Comment by Wolfteam** [comment]
@therealcorwin I did some updates on the general and main files (the new characters, weapons, etc).
For now, I'm gonna upload a beta version with your changes, should be up in 2hrs ~.

To join the beta program, go to the play store > genshindb > join beta program > wait 5 min. > go back and you should see an update

**Comment by therealcorwin** [comment]
Hi @Wolfteam,

I use the beta version (1.1.5+108 Beta). 

First, when i update the app to the beta version, i can't choose French in setting's menu.  I needed to delete cache/data. At first launch all the app and game data are in french. But if a want to swtich to english i can't roll back in french because i haven't this choice. Is't normal ? 

About translation, No big problem. Just fix some presentation, formulation, too long text. For exemple, some weapon's name are too long, can i use "\n" to write this on two line instead of one, or not ?

**Comment by Wolfteam** [comment]
> Hi @Wolfteam,
> 
> I use the beta version (1.1.5+108 Beta).
> 
> First, when i update the app to the beta version, i can't choose French in setting's menu. I needed to delete cache/data. At first launch all the app and game data are in french. But if a want to swtich to english i can't roll back in french because i haven't this choice. Is't normal ?
> 
> About translation, No big problem. Just fix some presentation, formulation, too long text. For exemple, some weapon's name are too long, can i use "\n" to write this on two line instead of one, or not ?

Hiho, gonna push a fix for the problem now, should be up in 2hrs~.

Yep you can use a line break to format the text.

**Comment by therealcorwin** [comment]
Version 109 beta tested. French option is available ;) 

Is there an easy solution in vscode or other to generate an APK with my translation to look at directly "in live" the modification on my computer/tablet ? 

I'll do some correction this weekend and V2 data if i have enough time :)

**Comment by Wolfteam** [comment]
> Version 109 beta tested. French option is available ;)
> 
> Is there an easy solution in vscode or other to generate an APK with my translation to look at directly "in live" the modification on my computer/tablet ?
> 
> I'll do some correction this weekend and V2 data if i have enough time :)

Yes, just follow the instructions that are located in the [README (https://github.com/Wolfteam/GenshinDb/blob/main/README.md) file.

After that, you should be able to run the app with one of the following commands:

Desktop:
``flutter run -d windows``

Android:
``flutter run``

**Pull request merged**